cmd/jujud/agent/machine_test.go

```diff
@@ -650,29 +649,65 @@ func (s *MachineSuite) TestManageModelAuditsAPI(c *gc.C) {
 		Password: password,
 	})
 
+	err := s.State.UpdateControllerConfig(map[string]interface{}{
+		"audit-log-exclude-methods": []interface{}{"Client.FullStatus"},
+	}, nil)
+	c.Assert(err, jc.ErrorIsNil)
+
 	s.assertJobWithState(c, state.JobManageModel, func(conf agent.Config, agentState *state.State) {
 		logPath := filepath.Join(conf.LogDir(), "audit.log")
 
-		apiInfo, ok := conf.APIInfo()
-		c.Assert(ok, jc.IsTrue)
-		apiInfo.Tag = user.Tag()
-		apiInfo.Password = password
-		st, err := api.Open(apiInfo, fastDialOpts)
-		c.Assert(err, jc.ErrorIsNil)
-		defer st.Close()
+		makeAPIRequest := func(doRequest func(*api.Client)) {
+			apiInfo, ok := conf.APIInfo()
+			c.Assert(ok, jc.IsTrue)
+			apiInfo.Tag = user.Tag()
+			apiInfo.Password = password
+			st, err := api.Open(apiInfo, fastDialOpts)
+			c.Assert(err, jc.ErrorIsNil)
+			defer st.Close()
+			doRequest(st.Client())
+		}
 
-		client := st.Client()
-		_, err = client.AddMachines([]params.AddMachineParams{{
-			Jobs: []multiwatcher.MachineJob{"JobHostUnits"},
-		}})
-		c.Assert(err, jc.ErrorIsNil)
+		// Make requests in separate API connections so they're separate conversations.
+		makeAPIRequest(func(client *api.Client) {
+			_, err = client.Status(nil)
+			c.Assert(err, jc.ErrorIsNil)
+		})
+		makeAPIRequest(func(client *api.Client) {
+			_, err = client.AddMachines([]params.AddMachineParams{{
+				Jobs: []multiwatcher.MachineJob{"JobHostUnits"},
+			}})
+			c.Assert(err, jc.ErrorIsNil)
+		})
 
-		// Check that there's a call to Client.AddMachinesV2 in the log.
+		// Check that there's a call to Client.AddMachinesV2 in the
+		// log, but no call to Client.FullStatus.
 		records := readAuditLog(c, logPath)
 		c.Assert(records, gc.HasLen, 3)
 		c.Assert(records[1].Request, gc.NotNil)
 		c.Assert(records[1].Request.Facade, gc.Equals, "Client")
 		c.Assert(records[1].Request.Method, gc.Equals, "AddMachinesV2")
+
+		// Now update the controller config to remove the exclusion.
+		err := s.State.UpdateControllerConfig(map[string]interface{}{
+			"audit-log-exclude-methods": []interface{}{},
+		}, nil)
+		c.Assert(err, jc.ErrorIsNil)
+		// Wait until the controller config change is propagated to
+		// the apiserver.
+		// TODO(babbageclunk): do this better!
```

Comment:
We can put this is an attempt strategy poll loop - there's existing examples of this in various tests